pkg/fakerp/config.go

```diff
@@ -8,30 +8,50 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/ghodss/yaml"
+
 	"github.com/openshift/openshift-azure/pkg/api"
+	pluginapi "github.com/openshift/openshift-azure/pkg/api/plugin/api"
 	"github.com/openshift/openshift-azure/pkg/fakerp/shared"
 	"github.com/openshift/openshift-azure/pkg/tls"
 )
 
 const (
-	LoggingSecretsDirectory = "secrets/"
+	SecretsDirectory      = "secrets/"
+	PluginConfigDirectory = "pluginconfig/"
+	TemplatesDirectory    = "/test/templates/"
 )
 
 func GetPluginConfig() (*api.PluginConfig, error) {
 	tc := api.TestConfig{
-		RunningUnderTest:      os.Getenv("RUNNING_UNDER_TEST") == "true",
-		ImageResourceGroup:    os.Getenv("IMAGE_RESOURCEGROUP"),
-		ImageResourceName:     os.Getenv("IMAGE_RESOURCENAME"),
-		DeployOS:              os.Getenv("DEPLOY_OS"),
-		ImageOffer:            os.Getenv("IMAGE_OFFER"),
-		ImageVersion:          os.Getenv("IMAGE_VERSION"),
-		ORegURL:               os.Getenv("OREG_URL"),
-		EtcdBackupImage:       os.Getenv("ETCDBACKUP_IMAGE"),
-		AzureControllersImage: os.Getenv("AZURE_CONTROLLERS_IMAGE"),
+		RunningUnderTest:   os.Getenv("RUNNING_UNDER_TEST") == "true",
+		ImageResourceGroup: os.Getenv("IMAGE_RESOURCEGROUP"),
+		ImageResourceName:  os.Getenv("IMAGE_RESOURCENAME"),
+	}
+
+	return &api.PluginConfig{
+		AcceptLanguages: []string{"en-us"},
+		TestConfig:      tc,
+	}, nil
+}
+
+func GetPluginTemplate() (*pluginapi.Config, error) {
+	// read template file without secrets
+	artifactDir, err := shared.FindDirectory(PluginConfigDirectory)
+	if err != nil {
+		return nil, err
+	}
+	data, err := readFile(filepath.Join(artifactDir, "pluginconfig-311.yaml"))
```

Comment:
This doesn't look right to me. Shouldn't we map plugin configs to versions and pick the correct config based on the version we are deploying?

Reply:
we had this discussion with @jim-minter. If I got this right, the plan is to maintain a minimal number of files `311, 4.0` and managed them via history/tags. More architecture topic discussion, not relevant for now as we dont know yet full implementation details